src/lib/mina_base/snapp_generators.ml

```diff
@@ -441,9 +441,10 @@ let gen_party_body (type a b) ?account_id ?balances_tbl ?(new_account = false)
   let%bind sequence_events =
     field_array_list_gen ~max_array_len:4 ~max_list_len:6
   in
-  let%map call_data = Snark_params.Tick.Field.gen in
+  let%bind call_data = Snark_params.Tick.Field.gen in
   (* update the depth when generating `other_parties` in Parties.t *)
   let depth = 0 in
+  let%map protocol_state = Snapp_predicate.Protocol_state.gen in
```

Comment:
This will (usually) fail, since the predicate will be random. Should we use the logic we have for the main predicate instead?